### ISSUE TITLE
Parentheses in nullish coalescing chains impacts type narrowing

### ISSUE THREAD

**Issue by joealden** [microsoft/TypeScript#44988]
# Bug Report

### 🔎 Search Terms

nullish coalescing chain parentheses type narrowing prettier

### 🕗 Version & Regression Information

This is the behavior in every version I tried, and I reviewed the FAQ for entries about nullish coalescing. Note that at the time of writing, 4.3.5 was the latest stable release.

### ⏯ Playground Link

[Playground link with relevant code](https://www.typescriptlang.org/play?ts=4.3.5#code/C4TwDgpgBAKlC8AoKUA+UDeUCGAuKAzsAE4CWAdgOYDcUARvkWVVAL7JqY6MkU30B+fAFdyAEwgAzChDFsO6LNiFRRE6eVm0GhXi1bVEiAMYB7ckSiUIwAEoQCwgDbAAjAigAKAG7YnwiFceZkpONSkZMQAaKF9-CAAmfBgASmC+BAA+TA5iG2Ficli-APcBAWL4hIA6bChyyoCaukMDIzMLYCsbe0cXBI8fEsD0lnRwjVkYuKbktN0QrJyUPOACopnA+oqhqtrtxsTquhTW6iA)

### 💻 Code

```ts
type T =
  | { a: string; b: string }
  | { a: string; b?: undefined }
  | { a?: undefined; b: string };

const getResult1 = (value1: string | undefined, value2: T): string => {
  return value1 ?? value2.a ?? value2.b;
};

const getResult2 = (value1: string | undefined, value2: T): string => {
  return value1 ?? (value2.a ?? value2.b);
};
```

### 🙁 Actual behavior

- `getResult1` throws a type error, as TypeScript incorrectly infers that `value2.b` has a type of `string | undefined`. 
- `getResult2` does not throw a type error, as TypeScript correctly infers that `value2.b` has a type of `string`. 

### 🙂 Expected behavior

The existence and placement of parentheses in nullish coalescing chains should not impact type narrowing, as to my knowledge (and `prettier`, as it automatically removes the parentheses seen in `getResult2`), `x ?? y ?? z` and `x ?? (y ?? z)` actually behave the same at runtime.

**Comment by MartinJohns** [comment]
Related: #42203

> Narrowings only occur on predefined syntactic patterns, [...]
